### DISCUSSION TITLE
Best practice way to fix `__dirname` issue in Server Components

### DISCUSSION THREAD

**Discussion by alexandros-megas** [vercel/next.js#63334] [App Router]
Hi all,

Our team recently ran into this issue: https://github.com/vercel/next.js/issues/49783. Specifically, we have a library that is using `@grpc/proto-loader` to load some protobuf files and provide users a gRPC API client.

https://github.com/Clarifai/clarifai-nodejs-grpc/blob/master/src/index.js#L15-L32

We noticed that specific issue is fixed by https://github.com/vercel/next.js/pull/49938 where a list of package names is maintained and excluded from bundling. On the other hand, [documentation suggests](https://nextjs.org/docs/app/api-reference/next-config-js/serverComponentsExternalPackages) that this is a temporary fix until those "popular packages" can update their code to be compatible with bundling.

So the question is: If our package has this issue, should we add the name to the excludes list, or should we work on a patch that fixes the bundling problem. And if we're expected to change our code, please provide some guidance on what patterns to use/avoid.

Also, this might be a dumb question, but what does the bundling process do that breaks things like __dirname? I also thought about adding these files to the `files` field in `package.json` but that is relevant to the package publish process, not bundling.

TL;DR: Some guidance on how to fix the `__dirname` issue would be appreciated.
